homeassistant/components/nanoleaf/config_flow.py

```diff
@@ -0,0 +1,226 @@
+        except NotAuthorizingNewTokens:
+            return self.async_show_form(
+                step_id="link", errors={"base": "not_allowing_new_tokens"}
+            )
+        except Unavailable:
+            return self.async_show_form(
+                step_id="user",
+                data_schema=USER_SCHEMA,
+                errors={"base": "cannot_connect"},
+                last_step=False,
+            )
+        except Exception:  # pylint: disable=broad-except
+            _LOGGER.exception("Unknown error authorizing Nanoleaf")
+            return self.async_show_form(step_id="link", errors={"base": "unknown"})
+        return await self.async_setup_finish()
+
+    async def async_step_import(self, config: dict[str, Any]) -> FlowResult:
+        """Handle Nanoleaf configuration import."""
+        self._async_abort_entries_match({CONF_HOST: config[CONF_HOST]})
+        _LOGGER.debug(
+            "Importing Nanoleaf on %s from your configuration.yaml", config[CONF_HOST]
+        )
+        self.nanoleaf = Nanoleaf(config[CONF_HOST])
+        self.nanoleaf.token = config[CONF_TOKEN]
+        return await self.async_setup_finish()
+
+    async def async_setup_finish(self) -> FlowResult:
+        """Finish Nanoleaf config flow."""
+        try:
+            info = await self.hass.async_add_executor_job(
+                pynanoleaf_get_info, self.nanoleaf
+            )
+        except Unavailable:
+            return self.async_show_form(
+                step_id="user",
+                data_schema=USER_SCHEMA,
+                errors={"base": "cannot_connect"},
+                last_step=False,
+            )
+        except InvalidToken:
+            return self.async_show_form(
+                step_id="link", errors={"base": "invalid_token"}
+            )
+        except Exception:  # pylint: disable=broad-except
+            _LOGGER.exception(
+                "Unknown error connecting with Nanoleaf at %s with token %s",
+                self.nanoleaf.host,
+                self.nanoleaf.token,
+            )
+            return self.async_show_form(
+                step_id="user",
+                data_schema=USER_SCHEMA,
+                errors={"base": "unknown"},
+                last_step=False,
+            )
+        name = info["name"]
+
+        await self.async_set_unique_id(name)
+        self._abort_if_unique_id_configured({CONF_HOST: self.nanoleaf.host})
+
+        # Show successfully imported instructions and optionally update the Nanoleaf config file
+        if self.context["source"] == config_entries.SOURCE_IMPORT:
+            persistent_notification.async_create(
+                self.hass,
+                f"Nanoleaf {name} has been successfully imported. Please remove the light with host {self.nanoleaf.host} from your configuration.yaml.",
+                f"Imported Nanoleaf {name}",
+            )
+        elif self.context["source"] == config_entries.SOURCE_INTEGRATION_DISCOVERY:
```

Comment:
```suggestion
        if self.context["source"] == config_entries.SOURCE_INTEGRATION_DISCOVERY:
```

We don't generate persistent notification on config entry import. It should be listed as a breaking change in the release notes intead.